client/src/main/java/com/netflix/conductor/client/automator/TaskPollExecutor.java

```diff
@@ -357,4 +398,31 @@ private PollingSemaphore getPollingSemaphore(String taskType) {
             return pollingSemaphoreMap.get(ALL_WORKERS);
         }
     }
+
+    private Runnable extendLease(Task task, CompletableFuture<Task> taskCompletableFuture) {
+        return () -> {
+            if (taskCompletableFuture.isDone()) {
+                LOGGER.warn(
+                        "Task processing for {} completed, but its lease extend was not cancelled",
+                        task.getTaskId());
+                return;
+            }
+            LOGGER.info("Attempting to extend lease for {}", task.getTaskId());
+            try {
+                TaskResult result = new TaskResult(task);
+                result.getLogs().add(new TaskExecLog("lease extend"));
```

Comment:
Since these logs are meant for user to add info about task execution, it might be detrimental to add this line here since this would add a lot of noise.
I would suggest changing this to a ```LOGGER.debug()``` statement. This would also improve the performance of this API as it only needs to update primary datastore.